lib/plugins/filter/before_post_render/backtick_code_block.ts

```diff
@@ -14,13 +17,44 @@ interface Options {
   firstLineNumber?: string | number
 }
 
+class CodeBlockEscape {
+  public stored: string[];
+
+  constructor() {
+    this.stored = [];
+  }
+
+  static escapeContent(cache: string[], flag: string, str: string) {
+    return `<!--${flag}\uFFFC${cache.push(str) - 1}-->`;
+  }
+
+  static restoreContent(cache: string[]) {
+    return (_: string, index: number) => {
+      assert(cache[index]);
+      const value = cache[index];
+      cache[index] = null;
+      return value;
+    };
+  }
+
+  restoreComments(str: string) {
+    return str.replace(rCommentHolder, CodeBlockEscape.restoreContent(this.stored));
+  }
+
+  escapeComments(str: string) {
+    return str.replace(rCommentEscape, (_, content) => CodeBlockEscape.escapeContent(this.stored, 'comment', content));
+  }
```

Comment:
It looks a bit weird. Your code repeatedly escape and restore comments in 'post' and 'backtick_code_block'. 

Is it possible to work like 'backtick_code_block', escape comment before 'backtick_code_block', restore them at the end of post render.(I am assume that the comments should not be rendered at all. I am not sure if it is right.)

Reply:
This means we need to share the escaped cache on a per-post basis between `before_post_render` and `post_render` (either by mounting a `stored` on each post or maintaining a global map). I'm not sure if this will increase the complexity of the code, and whether holding the cache for a long time will affect garbage collection. However, regardless, I will give it a try.